src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,492 @@
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.CaViewCasesTab
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.ACTIVE
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.APPROVED
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.IN_PROGRESS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.NOT_STARTED
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.OOS_BOTUS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.OOS_RECALL
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.SUBMITTED
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.TIMED_OUT
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_APPROVED
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_IN_PROGRESS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_SUBMITTED
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.isBreachOfTopUpSupervision
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.isEligibleEDS
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.isParoleEligible
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.isRecall
+import java.time.Clock
+import java.time.LocalDate
+
+@Service
+class CaCaseloadService(
+  val caseloadService: CaseloadService,
+  private val clock: Clock,
+  private val probationSearchApiClient: ProbationSearchApiClient,
+  private val licenceService: LicenceService,
+  private val prisonApiClient: PrisonApiClient,
+  private val communityApiClient: CommunityApiClient,
+) {
+  fun getPrisonOmuCaseload(prisonCaseload: List<String>?, searchString: String?): CaCaseLoad {
+    val statuses = listOf(
+      APPROVED,
+      SUBMITTED,
+      IN_PROGRESS,
+      TIMED_OUT,
+      ACTIVE,
+    )
+    val filteredPrisons = prisonCaseload?.filterNot { it == "CADM" }
+    val existingLicences = licenceService.findLicencesMatchingCriteria(
+      LicenceQueryObject(
+        statusCodes = statuses,
+        prisonCodes = filteredPrisons,
+      ),
+    )
+    val eligibleExistingLicences = filterAndFormatExistingLicences(existingLicences)
+
+    val eligibleNotStartedLicences = findAndFormatNotStartedLicences(
+      existingLicences,
+      prisonCaseload,
+    )
+
+    if (eligibleExistingLicences.isEmpty() && eligibleNotStartedLicences.isEmpty()) {
+      return CaCaseLoad(cases = emptyList(), showAttentionNeededTab = false)
+    }
+
+    val cases = mapCasesToComs(eligibleExistingLicences + eligibleNotStartedLicences)
+
+    return buildCaseload(cases, searchString, "prison")
+  }
+
+  fun getProbationOmuCaseload(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**suggestion:** ‏I would have possibly split this out into a separate service as this is quite a hefty service already.

Reply:
will split this down.

Reply:
I think it is kind of creating more logical complexity.
So better to be in one service.